server/tests/runs.e2e-spec.ts

```diff
@@ -0,0 +1,341 @@
+        it('should respond with a list of runs', async () => {
+            const res = await get('runs', 200);
+            expects(res);
+            expect(res.body.totalCount).toBeGreaterThan(1);
+            expect(res.body.returnCount).toBeGreaterThan(1);
+        });
+
+        it('should respond with a list of runs with take parameter', () => takeTest('runs', expects));
+
+        it('should respond with a list of runs with skip parameter', () => skipTest('runs', expects));
+
+        it('should respond with list of runs filtered by mapID parameter', async () => {
+            const res = await get('runs', 200, { mapID: map1.id });
+            expects(res);
+            expect(res.body.totalCount).toBe(2);
+            expect(res.body.returnCount).toBe(2);
+            expect(res.body.response[0].mapID).toBe(map1.id);
+        });
+
+        it('should respond with a list of runs filtered by userID parameter', async () => {
+            const res = await get('runs', 200, { userID: user1.id });
+            expects(res);
+            expect(res.body.totalCount).toBe(2);
+            expect(res.body.returnCount).toBe(2);
+            expect(res.body.response[0].playerID).toBe(user1.id);
+        });
+
+        it('should respond with a list of runs filtered by a list of run ids', async () => {
+            const ids: String = user1.id + ',' + user2.id;
+            const res = await get('runs', 200, { userIDs: ids });
+            expects(res);
+            expect(res.body.totalCount).toBe(4);
+            expect(res.body.returnCount).toBe(4);
+            expect(ids).toContain(res.body.response[0].playerID.toString());
+        });
+
+        it('should respond with a list of runs filtered by flags', async () => {
+            const res = await get('runs', 200, { flags: 1 << 5 });
+            expects(res);
+            expect(res.body.totalCount).toBe(1);
+            expect(res.body.response[0].flags).toBe(run4.flags); // TODO: Change this to bitwise math later
+        });
+
+        it('should respond with a list of runs with the map include', () => expandTest('runs', expects, 'map', true));
+
+        it('should respond with a list of runs with the player include', () => expandTest('runs', expects, 'player', true));
+
+        it('should respond with a list of runs with the rank include', () => expandTest('runs', expects, 'rank', true));
+
+        it('should respond with a list of runs with the zoneStats include',  () => expandTest('runs', expects, 'zoneStats', true));
+
+        it('should respond with a list of runs with the baseStats include', () => expandTest('runs', expects, 'baseStats', true));
+
+        it('should respond with a list of runs with the mapWithInfo include', async () => {
+            const res = await get('runs', 200, { expand: 'mapWithInfo' });
+            expects(res);
+            res.body.response.forEach(x => {
+                expect(x.map.info).not.toBeNull();
+            });
```

Comment:
I prefer we use `expect(x.map).toHaveProperty('info')` here, mainly just to keep consistency with other tests. Also the curly braces are unnecessary!